components/automate-ui/src/app/modules/infra-proxy/json-treeable/json-treeable.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, Input, ElementRef, OnChanges } from '@angular/core';
+import * as jsonTree from '../../../../app/page-components/json-tree/vendor/json-tree';
+
+// TODO:eng-ex: This might do better as a service or module.
+//              Unclear because there is styling on the javascript but no template.
+@Component({
+  selector: 'app-json-treeable',
+  template: '<div id="tree-container"></div>'
+})
+export class JsonTreeableComponent implements OnChanges {
+
+  @Input() json: Object;
+  tree: any;
+
+  constructor(private el: ElementRef) {}
+
+  ngOnChanges() {
+    this.build();
+  }
+
+  expand() {
+    if (this.tree) {
+      this.tree.expand();
+    }
+  }
+
+  collapse() {
+    if (this.tree) {
+      this.tree.collapse();
+    }
+  }
+
+  search(term): number {
+    let resultCount = 0;
+    this.reset();
+
+    function parentsOf(node) {
+      const parents = [];
+      let p = node.parentNode;
+
+      while (p) {
+        const o = p;
+        parents.push(o);
+        p = o.parentNode;
```

Comment:
Could be simplified:
```suggestion
        parents.push(p);
        p = p.parentNode;
```